### DISCUSSION TITLE
Allow Editing of Playlist Thumbnail.

### DISCUSSION THREAD

**Discussion by Gtwizzy** [music-assistant/support#2712] [Feature requests and ideas]
I use Music Assistant currently for a managing a lot of Local Music. So I will often put together music from multiple local file folders for say a "cooking" playlist or "blazing" playlist. It would be great to have the ability to edit the Playlist Thubnails so that it makes it easy to search through your user created playlists at a clance if you're looking at a card view for example. Currently new playlists seem to just inherit 2 characters from the naming convention, many of which don't make sense as is the case for my current "blazin' " playlist is it just named B'.  

I'd love to see this feature added in future releases. Or even just having the Playlist inherit a piece of cover art from the first artist in the playlist would be a good start if custom playlist thumbnails is too difficult.

**Comment by OzGav** [comment]
Actually the thumbnail is supposed to be a collage of the albums that make it up so there might be a problem (This is a small test playlist I have)

![image](https://github.com/user-attachments/assets/2d7ae3bf-3cef-421e-969f-17f39c1d0150)

**Reply by Gtwizzy** [reply]
Can I ask how was this playlist created? Was it created as a Music Assistant Playlist, Spotify Playlist or Filesystem playlist? Additionally is any of the music in the playlist coming from the filesystem provider? Because anytime I create a playlist using local music (regardless of whether I create the playlist as a Music Assistant Playlist or a Filesystem Playlist) it doesn't not have this behavior.  If create a new playlist in Spotify or I have a preconfigured playlist in _from_ Spotify for instance it will create this kind of thumbnail. But no other version of creating a playlist allows for this.

However even with that aside. It would still be nice to be able to create or upload custom playlist artwork. As this would allow for a kids playlist for example that has Car ride songs on it with a picture of a car as the custom thumbnail to make it easy for someone to easily identify when scrolling through. Or for kids to see if they have their own dashboard that this one is the car songs one and this one is the ABC's.

This isn't what I want this functionality for. But I think you catch my drift.

**Reply by OzGav** [reply]
MA just automatically creates the thumbnail. At least that is what is supposed to happen. I am not sure if it is working with local provider playlists.
Your request is noted though. Will see how many votes it gets and how it fits in with other priorities.

**Reply by Gtwizzy** [reply]
I guess this is the main reason I'm looking for MA to allow for a feature where a custom or manual Thumbnail can be added similarly to the option that Spotify allows for to do this same thing

**Reply by OzGav** [reply]
FYI the auto thumbnail creation is working 
![image](https://github.com/user-attachments/assets/be6c7f16-7f9d-4e54-a757-20db442904c4)

**Comment by mwdle** [comment]
> I use Music Assistant currently for a managing a lot of Local Music. It would be great to have the ability to edit the Playlist Thumbnails so that it makes it easy to search through your user created playlists at a glance if you're looking at a card view for example.

I think this is a great idea. I use .m3u playlists on my local filesystem and while the auto generated covers are nice, I would like to use custom covers for my playlists.

I've seen discussions on some other music platforms like Navidrome suggesting [some potential solutions](https://github.com/navidrome/navidrome/issues/406#issuecomment-973101061) to this problem.

One solution that I think could work nicely is an extended m3u tag (seeing as m3u has no formal specfication) that allows users to specify a path to a custom cover image.

Alternatively just naming an image the same name as the playlist would suffice.

Of course if the developers decided to implement a cover art selection mechanism on the frontend that would also be great!

**Reply by marcelveldt** [reply]
what about we just look if a jpg/png file exists with the exact same name as the playlist ? That is super easy to implement

**Reply by mwdle** [reply]
> what about we just look if a jpg/png file exists with the exact same name as the playlist ? That is super easy to implement

That would be wonderful! Definitely choose whatever is easiest to implement... I'll be satisfied with any solution that allows me to select custom playlist art.

Thank you for all of your great work, this is an AMAZING product!!!

**Reply by mwdle** [reply]
Is this still being considered as a future addition to Music Assistant? It's unfortunate that not many people seem interested in a feature like this.

I may take a look at trying to implement this myself when I have the time and motivation if someone else doesn't get to it first. If someone could point me in the right direction for the code that handles the current thumbnail generation that would be appreciated!

**Reply by OzGav** [reply]
Not many votes so it isn't high on the list. If you are interested in contributing you should ask in Discord for help https://discord.com/channels/753947050995089438/1260489193763639338 and read here https://github.com/music-assistant/server/blob/dev/DEVELOPMENT.md

https://github.com/music-assistant/server/blob/7c6f40cd947a7ffc6f21b1cd9d55d85bb6c1f699/music_assistant/helpers/images.py

**Comment by OzGav** [comment]
From 2.9.0b10 MA will use an image file identically named to the playlist as the playlist thumb.

**Reply by ARTPro72** [reply]
Nice i love that i can use this now. Thanks!

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
